File: Core/Ideas.Models/Idea.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ideas.Models
{
    public static class Constants
    {
        public static string SendGridAPIKey = "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";
        public static string Site = "https://ideawebapp.azurewebsites.net";
        public static string Email = "support@QIdea.com";
        public static string Name = "QIdea Support";
        public static string CreateIdeaDesc = "New idea has been created by Creator";
        public static string EditIdeaDesc = "Idea has been edited by Creator";
        public static string WithdrawIdeaDesc = "Idea has been withdrawn by the Creator";
        public static string ApproveIdeaDesc = "Idea has been approved by the Moderator";
        public static string RejectIdeaDesc = "Idea has been rejected by the Moderator";
        public static string DeligateIdeaDesc = "Idea has been delegated to you for moderation";
        public static string PickIdeaDesc = "Idea has been picked by Picker";
        public static string PickIdeaDoneDesc = "Idea has been implemented by Picker";
        public static string PickIdeaGiveUpDesc = "Idea has been give up by Picker";
        public static string PickIdeaReworkDesc = "Idea has been asked to rework by Creator";
        public static string PickIdeaAcceptDesc = "Idea has been accepted by Creator";
        public static string PickIdeaReopenDesc = "Idea has been reopened by Moderator";
        public static string CommentIdeaDesc = "New comment has been added for Idea";
    }
    public enum NotificationType
    {
        CreateIdea = 1,
        UpdateIdea = 2,
        WithdrawIdea = 3,
        ApproveIdea = 4,
        RejectIdea = 5,
        DeligateIdea = 6,
        PickIdea = 7,
        PickIdeaDone = 8,
        PickIdeaGiveUp = 9,
        PickIdeaRework = 10,
        PickIdeaAccept = 11,
        PickIdeaReopen = 12,
        CommentIdea = 13,
        EditComment = 14,
        DeleteComment = 15
    }

    public enum ReceiverType
    {
        Approver = 1,
        Creator = 2,
        Picker = 3,
        Watcher = 4,
        Commenter = 5,
    }
    public class Idea : User
    {
        public string IdeaId { get; set; }
        public string Type { get; set; }
        public string IdeaName { get; set; }
        public string Title { get; set; }
        public string Source { get; set; }
        public string Description { get; set; }
        public string BusinessCase { get; set; }
        public string IdealTime { get; set; }
        public string BusinessJustification { get; set; }
        public string ContactName { get; set; }
        public string ContactEmail { get; set; }
        public string ContactMobileNo { get; set; }
        public string IdeaStatus { get; set; }
        public DateTimeOffset? CreatedDate { get; set; }
        public DateTimeOffset? UpdatedDate { get; set; }
        public int WatchCount { get; set; }
        public int CommentCount { get; set; }
        public string ModeratorId { get; set; }
        public string Moderator { get; set; }
        public string ModeratorEmail { get; set; }
        public string DelegatorId { get; set; }
        public string Delegator { get; set; }
        public string DelegatorEmail { get; set; }
        public string PickerId { get; set; }
        public string Picker { get; set; }
        public string PickerEmail { get; set; }
        public bool IsWatching { get; set; }
    }
}

File: Ideas.WebAngular/src/app/idea.ts
export class Request {
  public author: string;
  public ideaId: string;
  public assignee: string;
  public comments: string;
  public commentType: boolean;
  public commentParentId: number;
  public ideaPage: string;
  public pageSize: number;
  public currentPage: number;
  public orderBy: string;
  public order: string;
  public isWatching: boolean;
  public idea: Idea;
  public commentId?: string;
}

export class Response {
  public isSuccess: boolean;
  public message: string;
}

export class SignInResponse extends Response {
  public author: string;
  public role: string;
}

export class IdeasResponse extends Response {
  public ideas: Idea[];
}

export class WatchersResponse extends Response {
  public watchers: Watcher[];
}

export class CommentsResponse extends Response {
  public comments: Comment[];
}

export class AlertsResponse extends Response {
  public alerts: Alert[];
}

export class IdeaResponse extends Response {
  public idea: Idea;
}

export class User extends Response {
  public id: string;
  public name: string;
  public email: string;
  public role: string[];
}

export class Watcher extends User {
  public ideaId: string;
}

export class Alert extends User {
  public alertId: string;
  public ideaId: string;
  public ideaName: string;
  public alertType: string;
  public alertDescription: string;
  public createdDate: string;
  public alertFlag: boolean;
}

export class Dashboard {
  public totalIdeas: string;
  public ideasInAction: string;
  public ideasPendingAction: string;
  public ideasTrending: string;
  public mostCommented: Idea;
}

export class Comment extends User {
  public commentId: string;
  public ideaId: string;
  public parentCommentId: string;
  public commentDescription: string;
  public createdDate: string;
  public updatedDate: string;
  public isPublic: boolean;
}

export class Idea extends User {
  public ideaId: string;
  public type: string;
  public ideaName: string;
  public title: string;
  public source: string;
  public description: string;
  public businessCase: string;
  public idealTime: string;
  public businessJustification: string;
  public contactName: string;
  public contactEmail: string;
  public contactMobileNo: string;
  public ideaStatus: string;
  public createdDate: string;
  public updatedDate: string;
  public watchCount: number;
  public commentCount: number;
  public moderatorId: string;
  public moderator: string;
  public moderatorEmail: string;
  public delegatorId: string;
  public delegator: string;
  public delegatorEmail: string;
  public pickerId: string;
  public picker: string;
  public pickerEmail: string;
  public action?: string;
  public comments?: string;
  public assignee?: string;
  public watchers?: Watcher[];
  public commentList?: Comment[];
  public commentId?: string;
  public isWatching: boolean;
  public isChild?: boolean;
  public section?: string;
  public currentSection?: string;
}
